solr/core/src/java/org/apache/solr/handler/component/RealTimeGetComponent.java

```diff
@@ -355,8 +370,9 @@ void processGetInputDocument(ResponseBuilder rb) throws IOException {
 
     String idStr = params.get("getInputDocument", null);
     if (idStr == null) return;
+    BytesRef idBytes = req.getSchema().indexableUniqueKey(idStr);
     AtomicLong version = new AtomicLong();
-    SolrInputDocument doc = getInputDocument(req.getCore(), new BytesRef(idStr), version, null, Resolution.DOC);
+    SolrInputDocument doc = getInputDocument(req.getCore(), idBytes, idBytes, version, null, Resolution.ROOT_WITH_CHILDREN);
```

Comment:
Wondering about `req.getSchema()` vs. `req.getCore().getLatestSchema()` here - https://github.com/apache/solr/pull/980 opened to explore further.